tests/integration-tests/clusters_factory.py

```diff
@@ -417,10 +424,12 @@ def create_cluster(self, cluster, log_error=True, raise_on_error=True, **kwargs)
 
         # create the cluster
         logging.info("Creating cluster {0} with config {1}".format(name, cluster.config_file))
+        self.custom_cli_credentials = kwargs.get("custom_cli_credentials")
```

Comment:
Can you not just reference the credentials from the __init__ function?  Why reset this value here?

Reply:
Either we should define the credentials for the factory or use the credentials provided to the create method, but not do both

Reply:
I'm leaning toward the latter, since the factory can then be independent of the cluster's credentials and it seems that the factory only uses the credentials argument (and the run_pcluster_command) in the creation of the cluster

Reply:
Since the clusters_factory(scope class) fixture in conftest is calling the Clustersfactory and if I setup the custom_cli_credentials value there it will always take the default None value. As in test_iam class any class which comes first will setup the ClusterFactory object. So when I use cluster_factory fixture in my test case I would have to send my custom_credentials where it will reset the object member in my test case.

Reply:
I'm not sure I follow.  It doesn't make sense to me to do both of these things
```
self.custom_cli_credentials = custom_cli_credentials
...
self.custom_cli_credentials = kwargs.get("custom_cli_credentials")
```
It seems like you are setting the class variable and getting the context from the create_cluster arguments.  So, the first time you set self.custom_cli_credentials is essentially dead code since it's replaced by the second call.  I think in this case since the clusters factory fixture is creating the Cluster object directly and bypassing the ClustersFactory class (a design flaw IMO) we should just remove the credentials member variable from the ClustersFactory class and just use the credentials provided to the create_cluster method.

Reply:
okay